extensions/Maps/includes/Maps_Geocoder.php

```diff
@@ -111,7 +111,21 @@ public function hasAlias( $alias ) {
 	 * @return array or false
 	 */
 	public function geocode( $address ) {
-		$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+
+		// MAIN-6654: cache requests to this API because it has daily limits
+		global $wgMemc;
+		$key = wfMemcKey( "geocoder", $address );
+		$response = $wgMemc->get( $key );
+		if ( empty($response) ) {
+			$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+			// MAIN-6654: make sure we do not return or store a bad response because it causes errors later
+			if ( strpos( $response, "OVER_QUERY_LIMIT" ) !== false ) {
+				$response = false;
+			}
+			if ( $response !== false ) {
+				$wgMemc->set( $key, $response, 86400 * 30 );  // 30 days
```

Comment:
This is a third party extension, I'd prefer to avoid adding dependencies on wikia classes/constants.